blockchain/validate.go

```diff
@@ -752,14 +752,35 @@ func (b *BlockChain) checkBlockContext(block *btcutil.Block, prevNode *blockNode
 
 	fastAdd := flags&BFFastAdd == BFFastAdd
 	if !fastAdd {
+		// Obtain the latest state of the deployed CSV soft-fork in
+		// order to properly guard the new validation behavior based on
+		// the current BIP 9 version bits state.
+		csvState, err := b.deploymentState(prevNode, chaincfg.DeploymentCSV)
+		if err != nil {
+			return err
+		}
+
+		// Once the CSV soft-fork is fully active, we'll switch to
+		// using the current median time past of the past block's
+		// timestamps for all lock-time based checks.
+		blockTime := header.Timestamp
+		if csvState == ThresholdActive {
+			medianTime, err := b.calcPastMedianTime(prevNode)
```

Comment:
As I noted on IRC, I believe that this is actually _correct_. The MTP used for a block to decide transaction finality is the median of the _past_ 11 blocks. So this would mean using `prevNode` everywhere during transaction finality tests as a part of block validation. 

This meshes with the mempool behavior at it uses the current MTP (of the chain tip) to decide if the transaction can be accepted into the mempool based on its eligibility for the _next_ block. The latest set of commits includes a boundary condition tests for the adherence of MTP.